src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -36,12 +36,13 @@ public void doUndo(CellGrid currentCellGridFromBoard) {
 		 * revert last action on the sudokuActionQueue, and put that action into
 		 * sudokuActionQueueForUndo
 		 */
-		SudokuAction lastAction = sudokuActionQueue.pop();
 		
-		if (null != lastAction) {
-			CellGrid previousCellGrid = lastAction.getCellGrid();
-			lastAction.setCellGrid(new CellGrid(currentCellGridFromBoard));
-			sudokuActionQueueForUndo.add(lastAction);
+		if (null != sudokuActionQueue.peek()) {

```

Comment:
was going to use queue then switched to stack and the name got stuck. will change when I get a chance.